packages/api/tests/endpoints/test-executions.js

```diff
@@ -170,3 +369,240 @@ test('GET fails if execution is not found', async (t) => {
+    fakeExecutions.map(async (execution) => {
+      // delete async operation foreign key to avoid needing a valid async operation
+      delete execution.asyncOperationId;
+      await executionModel.create(execution);
+      const executionPgRecord = await translateApiExecutionToPostgresExecution(
+        execution,
+        knex
+      );
+      await executionPgModel.create(knex, executionPgRecord);
+    })
+  );
+
+  t.true(await executionModel.exists({ arn: newExecution.arn }));
+  t.true(await executionPgModel.exists(knex, { arn: newExecution.arn }));
+
+  await request(app)
+    .delete(`/executions/${newExecution.arn}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(200);
+
+  // Correct Dynamo and PG execution was deleted
+  t.false(await executionModel.exists({ arn: newExecution.arn }));
+
+  const dbRecords = await executionPgModel.search(t.context.knex, {
+    arn: newExecution.arn,
+  });
+
+  t.is(dbRecords.length, 0);
+
+  // Previously created executions still exist
+  t.true(await executionModel.exists({ arn: fakeExecutions[0].arn }));
+  t.true(await executionModel.exists({ arn: fakeExecutions[1].arn }));
+
+  const originalExecution1 = await executionPgModel.search(t.context.knex, {
+    arn: fakeExecutions[0].arn,
+  });
+
+  t.is(originalExecution1.length, 1);
+
+  const originalExecution2 = await executionPgModel.search(t.context.knex, {
+    arn: fakeExecutions[1].arn,
+  });
+
+  t.is(originalExecution2.length, 1);
+});
+
+test('DELETE returns a 404 if Dynamo execution cannot be found', async (t) => {
+  const nonExistantExecution = {
+    arn: 'arn9',
+    status: 'completed',
+    name: 'test_execution',
+  };
+
+  const response = await request(app)
+    .delete(`/executions/${nonExistantExecution.arn}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(404);
+
+  t.is(response.body.message, 'No record found');
+});
+
+test('GET /history/:granuleId returns all workflow names associated with the granule', async (t) => {
+  const { fakeGranules, fakePGExecutions } = t.context;
+
+  const response = await request(app)
+    .get(`/executions/history/${fakeGranules[0].granuleId}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`);
+
+  response.body.forEach((execution, index) => {
+    t.is(execution.arn, fakePGExecutions[index].arn);
+    t.is(execution.workflow_name, fakePGExecutions[index].workflow_name);
+  });
+});
+
+test('POST /history returns all workflow names when ids array is passed', async (t) => {
+  const { fakeGranules, fakePGExecutions } = t.context;
+
+  const response = await request(app)
+    .post('/executions/history')
+    .send({ ids: [fakeGranules[0].granuleId, fakeGranules[1].granuleId] })
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`);
+
+  response.body.forEach((execution, index) => {
```

Comment:
Are the same set of executions returned by this test and the previous one for a specific granule ID? If so, how do we know whether the logic for one granule ID vs multiple is working correctly?

Reply:
Adde an additional test execution so that the two tests are returning different sets